pkg/ring/ring.go

```diff
@@ -655,6 +659,43 @@ func (r *Ring) countTokens() (map[string]uint32, map[string]int64) {
 	return numTokens, owned
 }
 
+// autoForgetUnhealthy forgets unhealthy ingesters from the ring.
+func (r *Ring) AutoForgetUnhealthy(ctx context.Context) {
```

Comment:
Instead of adding `AutoForgetUnhealthy` here I prefer to reuse the existing `NewAutoForgetDelegate`.
To make that work, we need to call `i.delegate.OnRingInstanceHeartbeat(i, ctx)` inside `i.updateConsul` and pass the same parameters.

Reply:
+1

it should be similar to how https://github.com/cortexproject/cortex/blob/de5cfe1e9aad18f60e2a091c0cec34339ab7eaea/pkg/ring/basic_lifecycler_delegates.go#L143
works

Reply:
After talking to @eeldaly offline I realized that the parameter is a bit different so probably hard to add the same handler.

This seems cleaner to me. We can switch `AutoForgetDelegate` to call this function instead.

```
func AutoForget(ringDesc *Desc, forgetPeriod time.Duration, logger log.Logger) {
	for id, instance := range ringDesc.Ingesters {
		lastHeartbeat := time.Unix(instance.GetTimestamp(), 0)

		if time.Since(lastHeartbeat) > forgetPeriod {
			level.Warn(d.logger).Log("msg", "auto-forgetting instance from the ring because it is unhealthy for a long time", "instance", id, "last_heartbeat", lastHeartbeat.String(), "forget_period", forgetPeriod)
			ringDesc.RemoveIngester(id)
		}
	}
}
```